text/0000-api-version-compat.md

```diff
@@ -0,0 +1,121 @@
+- Start Date: 2020-05-14
+- Author(s): [Emily Casey](https://github.com/ekcasey)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+
+# Summary
+[summary]: #summary
+
+To enable smoother migrations and upgrades this RFC proposes the following changes to the API version compatibility rules:
+
+1. When the lifecycle increments a supported API in its descriptor file to `0.x`, this implies support for `0.2-0.x` versions of that API
+1. The spec may make changes it wishes in a new 0.x API so long as those changes do not preclude buildpacks implementing different 0.x APIs from running together (e.g changes to the contractual build plan that cannot be bridged by the lifecycle)
+1. The platform can configure which version of the API the lifecycle implements
+1. When interacting with a buildpack, the lifecycle will use the API declared in each buildpack's `buildpack.toml`
+
+The lifecycle will assume the burden of maintaining compatibility, reducing the burden on platform and buildpack authors.  
+
+# Motivation
+[motivation]: #motivation
+
+Because there are far fewer lifecycle implementation than platforms or buildpacks, the lifecycle should assume the burden of maintaining compatibility. This will free platform authors to consume new lifecycles without worry and free buildpack authors to implement new APIs at their own pace. 
+
+## Non-Breaking 0.x API Versions
+Originally we specified that all minor version increments in the `0.x` line would be interpreted as breaking changes when applying [compatibility rules](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpacktoml-toml).
+However, a breaking change in the spec need not imply a breaking change in the lifecycle.
+The existing compatibility rules have become annoying over time as they add considerable friction to specification changes. All platforms and buildpacks must explicitly update to support the new APIs before pulling in a new lifecycle.
+As the community of platform and buildpack authors grows and matures, this pain is magnified.
+
+Now that the lifecycle has matured and the APIs have partially stabilized, we can realistically guarantee that we will not make breaking lifecycle changes when adding support for a new 0.x API.
+
+## Deprecations
+Once the 1.0 API is supported, we may wish to deprecate older APIs. A future RFC should provide a strategy for deprecations and outline the upgrade path from 0.x to 1.0 APIs. This is purposefully omitted from this RFC so that we can make progress on new 0.x APIs while we work out the details.
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Spec
+The spec will use future `0.x` API releases to include any desired changes with one exception: the spec may not change in a way that precludes buildpacks implementing different 0.x APIs from running together.
+
+This RFC does not make changes to the API compatibility rules post 1.0. 1.x APIs version are stil assumed to be purely additive.
```

Comment:
What about 1.0 itself. Are we ok with precluding 0.x and 1.0 buildpacks from running together?